### ISSUE TITLE
v14-rc1 - setting tags datatypes Storage Type to Csv or Json, causes content page to not load or not save

### ISSUE THREAD

**Issue by KevinJump** [umbraco/Umbraco-CMS#16084]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

v14-rc1

### Bug summary

Settings the "Storage Type" value of the Tags datatype causes JSON errors either on load or on Save of a content page (it depends if you have already created the page or not). 

### Specifics

It looks like the functionality works if you don't change the 'storage type' setting because the default is a not set value, but once the value is set it breaks. 

### Steps to reproduce

- Create a Tags data type. 
- Save it
- Change the storage type to Csv 
- Save It
- Change it back to Json 
- Save it. 

- Create or view an existing page that uses a document type with the Tag data type on it.,

Page will either fail to load with the error: 

```
The configuration for data type 1041 : Umbraco.Tags is invalid (see inner exception). Please fix the configuration and ensure it is valid. The site may fail to start and / or load data types and run.
```

or when you save the page will hand, and the error below will be in the logs

```
 An unhandled exception has occurred while executing the request.
System.InvalidOperationException: The configuration for data type -87 : Umbraco.RichText is invalid (see inner exception). Please fix the configuration and ensure it is valid. The site may fail to start and / or load data types and run.
 ---> System.InvalidOperationException: Failed to parse configuration "System.Collections.Generic.Dictionary`2[System.String,System.Object]" as "RichTextConfiguration" (see inner exception).
 ---> System.FormatException: String "069ef55e-6a88-4deb-9d2c-60df36b6eece" is not a valid udi.
   at Umbraco.Cms.Core.UdiParser.ParseInternal(String s, Boolean tryParse, Boolean knownTypes, Udi& udi)
   at Umbraco.Cms.Core.UdiParser.Parse(String s)
   at Umbraco.Cms.Infrastructure.Serialization.JsonUdiConverter.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.ReadAsObject(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.Serialization.Converters.CastingConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
```

**Comment by elit0451** [comment]
Hi there @KevinJump 👋 

Thanks for reaching out! I can indeed reproduce this, we will have a look 🙂

**Comment by elit0451** [comment]
The error in the logs that you posted here seems to be a wrong copy/paste from another issue 😜  

Here is the error we get in the logs for this issue:
```
System.InvalidOperationException: The configuration for data type 1055 : Umbraco.Tags is invalid (see inner exception). Please fix the configuration and ensure it is valid. The site may fail to start and / or load data types and run.
 ---> System.InvalidOperationException: Failed to parse configuration "System.Collections.Generic.Dictionary`2[System.String,System.Object]" as "TagConfiguration" (see inner exception).
 ---> System.Text.Json.JsonException: The JSON value could not be converted to Umbraco.Cms.Core.Models.TagsStorageType. Path: $.storageType | LineNumber: 0 | BytePositionInLine: 16.
   at System.Text.Json.ThrowHelper.ThrowJsonException(String message)
   at System.Text.Json.Serialization.Converters.EnumConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 json, JsonTypeInfo`1 jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Umbraco.Cms.Infrastructure.Serialization.SystemTextJsonSerializerBase.Deserialize[T](String input) in /UmbracoCMS/src/Umbraco.Infrastructure/Serialization/SystemTextJsonSerializerBase.cs:line 17
   at Umbraco.Cms.Core.PropertyEditors.ConfigurationEditor`1.ToConfigurationObject(IDictionary`2 configuration, IConfigurationEditorJsonSerializer configurationEditorJsonSerializer) in /UmbracoCMS/src/Umbraco.Core/PropertyEditors/ConfigurationEditorOfTConfiguration.cs:line 39
   --- End of inner exception stack trace ---
   at Umbraco.Cms.Core.PropertyEditors.ConfigurationEditor`1.ToConfigurationObject(IDictionary`2 configuration, IConfigurationEditorJsonSerializer configurationEditorJsonSerializer) in /UmbracoCMS/src/Umbraco.Core/PropertyEditors/ConfigurationEditorOfTConfiguration.cs:line 43
   at Umbraco.Cms.Core.Models.DataType.get_ConfigurationObject() in /UmbracoCMS/src/Umbraco.Core/Models/DataType.cs:line 104
```

**Comment by leekelleher** [comment]
Cross-linking to PR for frontend fix: https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/1680
Tested the original scenario and the server is happy with a single `string` value now; no error thrown.

**Comment by leekelleher** [comment]
PR: https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/1680 has been merged in and will be part of upcoming v14.0.0-rc2.
